### PR TITLE
Integrate clone repo flow into gitness

### PR DESCRIPTION
**Pull request by sans-harness**
- Adds clone repo to `gitness`. 
- Adds create token flow to clone in repo-summary page. 

https://github.com/user-attachments/assets/92993506-66e0-483c-ae09-4ba364c15c18

### CODE REVIEW

File: apps/gitness/src/pages/repo-sandbox/repo-sandbox-summary.tsx [hunk 1]
```diff
@@ -176,93 +208,106 @@ export const RepoSandboxSummaryList: React.FC = () => {
...
+                    count: pull_req_summary?.open_count || 0,
+                    iconName: 'open-pr'
+                  }
+                ]}
+              />
+            </SandboxLayout.Content>
+          </SandboxLayout.Column>
+        </SandboxLayout.Columns>
+      </SandboxLayout.Main>
+      {createdTokenData && (
```
Comment (abhinavrastogi-harness) [inline apps/gitness/src/pages/repo-sandbox/repo-sandbox-summary.tsx:304]: why do we need this condition?

File: packages/playground/src/components/repo-clone/clone-repo-dialog.tsx [hunk 2]
```diff
@@ -2,18 +2,23 @@ import { CloneRepoForm } from './clone-repo-form'
...
 
-export const CloneRepoDialog = () => {
+export const CloneRepoDialog = ({
+  httpsUrl,
+  sshUrl,
+  handleCreateToken
+}: {
+  sshUrl: string
+  httpsUrl: string
+  handleCreateToken: () => void
```
Comment (abhinavrastogi-harness) [inline packages/playground/src/components/repo-clone/clone-repo-dialog.tsx:12]: let's pull this into an exported interface `CloneRepoDialogProps`. helps with debugging/autocomplete etc.

File: packages/playground/src/components/repo-clone/clone-repo-form.tsx [hunk 3]
```diff
@@ -14,7 +14,12 @@ const cloneRepoFormSchema = z.object({
 
 export type CloneRepoFormType = z.infer<typeof cloneRepoFormSchema>
 
-export function CloneRepoForm({ httpsUrl, sshUrl, currentTab }: CloneRepoFormType & { currentTab: string }) {
+export function CloneRepoForm({
+  httpsUrl,
+  sshUrl,
+  currentTab,
+  handleCreateToken
+}: CloneRepoFormType & { currentTab: string; handleCreateToken: () => void }) {
```
Comment (abhinavrastogi-harness) [inline packages/playground/src/components/repo-clone/clone-repo-form.tsx:22]: same here.. create an interface `CloneRepoFormProps` that's exported

File: apps/gitness/src/pages/repo-sandbox/repo-sandbox-summary.tsx [hunk 4]
```diff
@@ -176,93 +208,106 @@ export const RepoSandboxSummaryList: React.FC = () => {
...
-              <ListActions.Left>
-                <ButtonGroup.Root>
-                  <BranchSelector
-                    size="default"
-                    name={selectedBranch}
-                    branchList={branchList}
-                    selectBranch={branch => selectBranch(branch)}
-                  />
+    <>
+      <SandboxLayout.Main hasLeftPanel hasHeader hasSubHeader>
```
Comment (abhinavrastogi-harness) [inline apps/gitness/src/pages/repo-sandbox/repo-sandbox-summary.tsx:212]: ideally most of this structure should be in playground components. but we can take that up separately.

File: apps/gitness/src/pages/repo-sandbox/repo-sandbox-summary.tsx (hunk 1 above)
Comment (sans-harness) [reply apps/gitness/src/pages/repo-sandbox/repo-sandbox-summary.tsx:304]: in the case where `token-creation` failed, we do not open the dialog.

Comment (abhinavrastogi-harness) [reply apps/gitness/src/pages/repo-sandbox/repo-sandbox-summary.tsx:304]: In that case we should be able to do:
```
open={createdTokenData && successTokenDialog}
```

**Review by abhinavrastogi-harness — APPROVED** [review]

**Pull request merged**